### PR TITLE
Load port fix

### PR DESCRIPTION
**Pull request by afarooqui98**
cleaned up sym_object so it's hopefully easier to understand at first glance, separated backend variables and ui variables, and tried to make function descriptions visible

### CODE REVIEW

File: graphic_scene.py [hunk 1]
```diff
@@ -6,11 +6,22 @@
...
 from gui_views import state
 from sym_object import *
-from m5_calls import load_instantiate
 import string
 import random
+import collections
 import copy
 
+"""convert a dictionary with unicode keys and values to utf-8"""
+def convert(data):
```
Comment (powerjg) [inline graphic_scene.py:15]: Just out of curiosity... why is this needed? Why are the values "unicode" and not the correct type?

File: sym_object.py [hunk 2]
```diff
@@ -5,31 +5,55 @@
...
+import copy
 
 
 class SymObject(QGraphicsItemGroup):
 
     def __init__(self, x, y, width, height, scene, component_name, name,
                     loadingFromFile, state):
         super(SymObject, self).__init__()
+
+        """common variables"""
```
Comment (powerjg) [inline sym_object.py:17]: `"""` should only be used for docstrings, not for comments.

These should be normal python comments (`#`).

Let me know if I'm wrong (e.g., point me to what style you're following with this).

File: sym_object.py [hunk 3]
```diff
@@ -56,71 +80,144 @@ def __init__(self, x, y, width, height, scene, component_name, name,
         self.y = self.scenePos().y()
         self.state.current_sym_object = self
 
-    # Create an instantiated simobject for the symobject
+    """Get additional info on params such as default values  after
```
Comment (powerjg) [inline sym_object.py:83]: docstrings go after the definition. See PEP8 (https://www.python.org/dev/peps/pep-0008/#documentation-strings)

File: graphic_scene.py (hunk 1 above)
Comment (afarooqui98) [reply graphic_scene.py:14]: I think the way that the instance_ports are saved have to be in unicode, but I'll double check this. I know for a fact that the dictionary when loaded was being returned in unicode form, but maybe the way that we're saving might not be the most efficient.
